src/shared/Utilities/Util.cpp

```diff
@@ -736,3 +736,16 @@ void print_banner()
         break;
     }
 }
+
+// Sometimes needed whend using urand()
+void InvertMinMaxIfNeeded(uint32& min, uint32& max)
+{
+    uint32 minCopy = min;
+    uint32 maxCopy = max;
+
+    if (max < min)
+    {
+        min = maxCopy;
+        max = minCopy;
+    }
```

Comment:
This can be accomplished with one fewer copy.

copy = min;
if...
  min = max;
  max = copy;
}

Reply:
yeah you are right